File: 백준/Bronze/2747. 피보나치 수/피보나치 수.py
import sys

# 백준에서는 sys.stdin 사용, 로컬에서는 input.txt 파일 읽기
if sys.stdin.isatty():
    sys.stdin = open("../input.txt", "r")

# 입력 받기
n = int(input())  # 첫 번째 줄 (숫자 입력)

arr = [0] * (n+1)

arr[0] = 0
arr[1] = 1

for i in range(2,n+1):
  arr[i] = arr[i - 1] + arr[i - 2]
  
print(arr[n])

File: 백준/Bronze/2747. 피보나치 수/피보나치 수.js
/** 백준 https://www.acmicpc.net/problem/2747 */

const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "../input.txt";
const input = fs.readFileSync(filePath, "utf-8").trim().split("\n");

const n = parseInt(input);

const arr = new Array(n + 1).fill(0);

arr[0] = 0;
arr[1] = 1;
for (let i = 2; i < n + 1; i++) {
  arr[i] = arr[i - 1] + arr[i - 2];
}

/** 결과 출력 */
console.log(arr[n]);
